CMakeLists.txt

```diff
@@ -239,23 +239,18 @@ ENDIF()
 
 # UNIT TESTS VERIFYING CONSISTENT STATE OF OUR ROOT MACROS AND THE EXECUTION ENVIRONMENT
 if(CHECK_ROOTMACRO_COMPILE AND HAVESIMULATION)
-  FIND_PROGRAM(BASH_PROGRAM bash)
-
   # On Mac OS GLOB_RECURSE returns both .C and .c files, i.e. case insensitive
   FILE(GLOB_RECURSE MACRO_FILES "*.C")
   # Case sensitive filtering of .C files
   LIST(FILTER MACRO_FILES INCLUDE REGEX "^.*\\.C$")
   SET(TEST_NO 0)
   FOREACH(MACRO_FILE ${MACRO_FILES})
+    string(REPLACE ${CMAKE_SOURCE_DIR} "" MACRO_FILE_LABEL ${MACRO_FILE})
     MATH(EXPR TEST_NO "${TEST_NO} + 1")
-    SET(EXE_NAME ${CMAKE_BINARY_DIR}/tmp_scripts/macro_test${TEST_NO}.sh)
-
-    # Create test script for macros
-    FILE(WRITE ${EXE_NAME} "export LD_LIBRARY_PATH=$ENV{LD_LIBRARY_PATH}:${CMAKE_BINARY_DIR}/lib\n root -b -l -q -e \".L ${MACRO_FILE}\"")
-
-    ADD_TEST(NAME ${MACRO_FILE}
+    ADD_TEST(NAME ${MACRO_FILE_LABEL}
       WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-      COMMAND ${BASH_PROGRAM} ${EXE_NAME}
+      COMMAND root -b -l -q -e ".L ${MACRO_FILE}"
```

Comment:
So I would add `-n` to avoid executing ROOT logon/logoff scripts (we don't have them on our test nodes, but most people do have them on their own computers), plus I would also export `ROOT_HIST=0` to prevent `.L ${MACRO_FILE}` from being added to the user's ROOT history.

As in:

```
COMMAND env ROOT_HIST=0 root -b -l -n -q -e ".L ${MACRO_FILE}"
```

Reply:
OK, via `SET_PROPERTY(TEST ${MACRO_FILE_LABEL} PROPERTY ENVIRONMENT ...)` it is possible to do that without using `env`.

Reply:
I think I've addressed all your comments. I also added `DYLD_LIBRARY_PATH` to the environment, although according to Apple is not strictly necessary, because root might decide to behave differently.